src/pudl_archiver/orchestrator.py

```diff
@@ -186,25 +186,44 @@ async def run(self) -> RunSummary | Unchanged:
             await self._apply_change(deletion)
 
         new_datapackage, old_datapackage = await self._update_datapackage(resources)
-        run_summary = self.downloader.generate_summary(
-            old_datapackage, new_datapackage, resources
-        )
+        run_summary = self._summarize_run(old_datapackage, new_datapackage, resources)
 
         if not self.changes:
-            # must run *after* updating datapackage.json
-            logger.info("No changes detected.")
-            await self.depositor.delete_deposition(self.new_deposition)
-            return Unchanged(dataset_name=self.data_source_id)
+            # must run *after* potentially updating datapackage.json
+            logger.info(
+                "No changes detected, kept draft at "
+                f"{self.new_deposition.links.html} for inspection."
+            )
+            return run_summary
 
         if not run_summary.success:
-            logger.error("Archive validation failed. Not publishing new archive.")
-            await self.depositor.delete_deposition(self.new_deposition)
+            logger.error(
+                "Archive validation failed. Not publishing new archive, kept "
+                f"draft at {self.new_deposition.links.html} for inspection."
+            )
             return run_summary
 
         await self._publish()
-
         return run_summary
 
+    def _summarize_run(
```

Comment:
Moved this logic to `orchestrator` so that we wouldn't have to tell the archiver about the deposition links.